packages/react-router-with-query/src/index.tsx

```diff
@@ -180,3 +163,31 @@ export function routerWithQueryClient<TRouter extends AnyRouter>(
 
   return router
 }
+
+type PushableStream = {
+  stream: ReadableStream
+  enqueue: (chunk: unknown) => void
+  close: () => void
+  isClosed: boolean
+  error: (err: unknown) => void
+}
+function createPushableStream(): PushableStream {
+  let controllerRef: ReadableStreamDefaultController | undefined
+  const stream = new ReadableStream({
+    start(controller) {
+      controllerRef = controller
+    },
+  })
+  let isClosed = false
+
+  return {
+    stream,
+    enqueue: (chunk) => controllerRef?.enqueue(chunk),
+    close: () => {
+      controllerRef?.close()
+      isClosed = true
+    },
+    isClosed,
```

Comment:
Might be better to align and use a get function here (i.e `getIsClosedStatus()`)